backend/app/models/project_image.rb

```diff
@@ -0,0 +1,8 @@
+class ProjectImage < ApplicationRecord
+  belongs_to :project
+
+  has_one_attached :file
+
+  validates :file, attached: true, content_type: /\Aimage\/.*\z/, size: {less_than: 5.megabytes}
```

Comment:
I'm wondering how validation of maximum size works with DirectUpload. I think that might be part of frontend validation, they send metadata of the file, checksum, content-length and presigned url is given by the direct upload controller. So there should be no possibility of sending bigger file than presigned url is constructed for.

Reply:
I believe these validations are checked when association between active record blob and appropriate record is created, therefore it does not matter if you do direct upload or multipart one. At first case, you create blob first and then validation is triggered when ProjectImage is created (you provide signed_id as identification of already existing blob). At second case, you build and check validation at same time. Ofc when we are talking about direct upload, it is highly recommended to run same validations on FE side, otherwise you can create all blobs (and push all data to server) to just discover later on that files are not-valid...